provers/blevm/blevm-aggregator/src/main.rs

```diff
@@ -11,44 +11,34 @@ use buffer::Buffer;
 use sha2::{Digest, Sha256};
 
 pub fn main() {
-    // Read the number of proofs
-    let n: usize = sp1_zkvm::io::read();
-
-    if n < 2 {
-        panic!("must provide at least 2 proofs");
-    }
-
-    // Read all verification keys first
-    let mut verification_keys: Vec<[u32; 8]> = Vec::with_capacity(n);
-    for _ in 0..n {
-        verification_keys.push(sp1_zkvm::io::read());
-    }
-
-    // Read all public values
-    let mut public_values: Vec<Vec<u8>> = Vec::with_capacity(n);
-    for _ in 0..n {
-        public_values.push(sp1_zkvm::io::read());
-    }
-
-    // Verify all proofs using their respective verification keys
-    for (values, vk) in public_values.iter().zip(verification_keys.iter()) {
-        let proof_values_hash = Sha256::digest(values);
-        sp1_zkvm::lib::verify::verify_sp1_proof(vk, &proof_values_hash.into());
+    // Read the verification keys.
+    let vkeys = sp1_zkvm::io::read::<Vec<[u32; 8]>>();
+
+    // Read the public values.
+    let public_values = sp1_zkvm::io::read::<Vec<Vec<u8>>>();
+
+    // Verify the proofs.
+    assert_eq!(vkeys.len(), public_values.len());
+    for i in 0..vkeys.len() {
+        let vkey = &vkeys[i];
+        let public_values = &public_values[i];
+        let public_values_digest = Sha256::digest(public_values);
+        sp1_zkvm::lib::verify::verify_sp1_proof(vkey, &public_values_digest.into());
     }
 
     // Parse all outputs
-    let mut outputs: Vec<BlevmOutput> = Vec::with_capacity(n);
+    let mut outputs: Vec<BlevmOutput> = vec![];
     for values in &public_values {
         let mut buffer = Buffer::from(values);
         outputs.push(buffer.read::<BlevmOutput>());
     }
 
-    // Verify block sequence
-    for i in 1..n {
-        if outputs[i - 1].header_hash != outputs[i].prev_header_hash {
-            panic!("header hash mismatch at position {}", i);
-        }
-    }
+    // // Verify adjacent headers
```

Comment:
This fails as we don't have adjacent mock proofs and we're reusing the same proof twice. Working on updating mock to output two adjacent proofs.

Reply:
Fixed.